src/ImageSharp/Metadata/Profiles/IPTC/IptcProfile.cs

```diff
@@ -264,34 +268,34 @@ private void Initialize()
 
             this.values = new Collection<IptcValue>();
 
-            if (this.Data == null || this.Data[0] != 0x1c)
+            if (this.Data == null || this.Data[0] != IptcTagMarkerByte)
             {
                 return;
             }
 
-            int i = 0;
-            while (i + 4 < this.Data.Length)
+            int offset = 0;
+            while (offset + 4 < this.Data.Length)
             {
-                if (this.Data[i++] != 28)
+                bool isValidTagMarker = this.Data[offset++] == IptcTagMarkerByte;
+                byte recordType = this.Data[offset++];
+                var tag = (IptcTag)this.Data[offset++];
+
+                uint byteCount = BinaryPrimitives.ReadUInt16BigEndian(this.Data.AsSpan(offset, 2));
+                offset += 2;
+                if (byteCount > MaxStandardDataTagSize)
                 {
-                    continue;
+                    // Extended data set tag's are not supported.
+                    break;
                 }
 
-                i++;
-
-                var tag = (IptcTag)this.Data[i++];
-
-                int count = BinaryPrimitives.ReadInt16BigEndian(this.Data.AsSpan(i, 2));
-                i += 2;
-
-                var iptcData = new byte[count];
-                if ((count > 0) && (i + count <= this.Data.Length))
+                if (isValidTagMarker && byteCount > 0 && (offset + byteCount <= this.Data.Length))
```

Comment:
```suggestion
                if (isValidTagMarker && byteCount > 0 && (offset <= this.Data.Length - byteCount))
```
To prevent potential overflow. I see that this won't happen in practice, but I got conditioned to be paranoid about these cases (to be safe against malicious code, etc.).